test/integration/targets/aws_codepipeline/defaults/main.yml

```
@@ -6,4 +6,7 @@ codepipeline_name: "{{ resource_prefix }}-test-codepipeline"
 # IAM role names have to be less than 64 characters
 # The 8 digit identifier at the end of resource_prefix helps determine during
 # which test something was created and allows tests to be run in parallel
-codepipeline_service_role_name: "ansible-test-sts-{{ resource_prefix | regex_search('([0-9]+)$') }}-codebuild-service-role"
+# Shippable resource_prefixes are in the format shippable-123456-123, so in those cases
+# we need both sets of digits to keep the resource name unique
+unique_id: "{{ resource_prefix | regex_search('(\\d+-?)(\\d+)$') }}"
+codepipeline_service_role_name: "ansible-test-sts-{{ unique_id }}-codebuild-service-role"
```

Should the suffix be `-codepipeline-service-role`?
```suggestion
codepipeline_service_role_name: "ansible-test-sts-{{ unique_id }}-codepipeline-service-role"
```

---

Oh. Thank you, good catch. It was initially without the `-service` though https://github.com/ansible/ansible/blame/91ccb03552f52fe4dd15366f0b378f4e23cedeb9/test/integration/targets/aws_codepipeline/defaults/main.yml#L5.

```suggestion
codepipeline_service_role_name: "ansible-test-sts-{{ unique_id }}-codepipeline-role"
```